src/js/component/reportForm/index.js

```diff
@@ -0,0 +1,168 @@
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import Radio from '@mui/material/Radio';
+import RadioGroup from '@mui/material/RadioGroup';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import FormControl from '@mui/material/FormControl';
+import FormLabel from '@mui/material/FormLabel';
+import Button from '@mui/material/Button';
+import InputLabel from '@mui/material/InputLabel';
+import MenuItem from '@mui/material/MenuItem';
+import Select from '@mui/material/Select';
+import DownloadIcon from '@mui/icons-material/Download';
+import { withStyles } from '@mui/styles';
+import styles from './styles';
+import { post } from '../../axiosUtils/appUtils';
+import { useDispatch } from 'react-redux';
+import { manageToast, setOverlayStatus } from '../../store/common/actions';
+
+const ReportForm = (props) => {
+  const { classes } = props;
+  const dispatch = useDispatch();
+  const [reportFormat, setReportFormat] = useState('daily');
+  const [selectedDate, setSelectedDate] = useState(null);
+  const [selectedMonth, setSelectedMonth] = useState('');
+  const [selectedYear, setSelectedYear] = useState('');
+
+  const months = [
+    'January',
+    'February',
+    'March',
+    'April',
+    'May',
+    'June',
+    'July',
+    'August',
+    'September',
+    'October',
+    'November',
+    'December',
+  ];
+
+  const years = [
+    '2023',
+    '2022',
+    '2021',
+    '2020',
+    '2019',
+    '2018',
+    '2017',
+    '2016',
+    '2015',
+    '2014',
+    '2013',
+    '2012',
+  ];
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const formData = {
+      reportFormat,
+      selectedDate,
+      selectedYear,
+      selectedMonth,
+    };
+    console.log('formData', formData);
+
+    //TODO- replace with actual endpoint
+
+    // post('url', formData)
+    //   .then((res) => {})
+    //   .catch((error) => {
+    //     const obj = {
+    //       title: 'error',
+    //       message: error.message,
+    //       status: true,
+    //       type: 'error',
+    //     };
+    //     dispatch(manageToast(obj));
+    //     dispatch(setOverlayStatus(false));
+    //   });
+  };
+  return (
+    <form onSubmit={handleSubmit} className={classes.root}>
+      <FormControl sx={{ marginBottom: '10px' }}>
+        <FormLabel id='demo-controlled-radio-buttons-group'>
+          Pick the format of report:{' '}
+        </FormLabel>
+        <RadioGroup
+          aria-labelledby='demo-controlled-radio-buttons-group'
+          name='controlled-radio-buttons-group'
+          value={reportFormat}
+          onChange={(e) => setReportFormat(e.target.value)}
+        >
+          <FormControlLabel value='daily' control={<Radio />} label='Daily' />
+          <FormControlLabel
+            value='monthly'
+            control={<Radio />}
+            label='Monthly'
+          />
+        </RadioGroup>
+      </FormControl>
+      {reportFormat === 'daily' && (
```

Comment:
daily/ monthly ye sab static string ko constant file se uthaana sahi rahega for long term

Reply:
done